modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx

```diff
@@ -183,14 +184,28 @@ export default function ViewObjectDefinitions({
 			}
 
 			if (action.data.id === 'deleteObjectDefinition') {
-				deleteObjectDefinition({
-					baseResourceURL,
-					handleDeleteObjectDefinition: setDeletedObjectDefinition,
-					handleShowDeleteObjectDefinitionModal,
-					objectDefinitionId: itemData.id,
-					objectDefinitionName: itemData.name,
-					onAfterDeleteObjectDefinition: () => setReloadFDS(true),
-				});
+				if (
+					itemData.rootObjectDefinitionExternalReferenceCode &&
+					Liferay.FeatureFlags['LPS-187142']
+				) {
+					setSelectedObjectDefinition(itemData);
+
+					setShowModal((previousState) => ({
+						...previousState,
```

Comment:
This is weird to me... can you show more than one modal at a time? Why the previous state matters? Can we go from one modal to another? I know you did not create this pattern, it was there before, but I had to ask.